framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/deployment/OpenShiftDeployment.java

```diff
@@ -176,6 +180,13 @@ private Instance createInstance(Pod pod) {
     protected URL getHttpRouteUrl(String serviceName) {
         try {
             return getRouteUri(Protocol.http, serviceName).toURL();
+        } catch (RuntimeException e) {
+            if (e.getMessage().contains(Protocol.http + " route leading to service " + serviceName + " not found")) {
```

Comment:
Maybe it would be better just to return specific exception in case http route is not found.
Client code would have to take care of https retrieval by itself.